### ISSUE TITLE
DataKubernetesNamespace resource is missing name field

### ISSUE THREAD

**Issue by bartlettt** [hashicorp/terraform-cdk#856]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

Typescript

cdktf: 0.2.1
"@cdktf/provider-kubernetes": "^0.2.110"

### Affected Resource(s)

DataKubernetesNamespace

Probably other Kubernetes Data resources too

### Debug Output
https://gist.github.com/bartlettt/bd130b3e27063948a1a8677fcc6e7bd4

### Expected Behavior
Namespace name should have been printed to console

### Actual Behavior

Undefined was printed to console

### Steps to Reproduce

```
const dataNamespace = new DataKubernetesNamespace(stack, 'data-namespace', {
  metadata: [
    {
      name: 'kube-system'
    }
  ]
})

const namespaceName = dataNamespace.metadata[0].name
console.log(namespaceName)
```

**Comment by bartlettt** [comment]
Looks like a workaround is to use:
`const name = ns.metadataInput[0].name`

**Comment by danieldreier** [comment]
@bartlettt thanks for reporting this! I've reproduced this using CDK for Terraform 0.5 and the 2.3.2 (current) version of the `hashicorp/kubernetes` provider. This does look like it should work.

For the sake of the engineer who picks this up, I created a [simple reproduction case](https://github.com/danieldreier/terraform-cdk-issue-reproductions/tree/main/856) that can be git cloned.

**Comment by jsteinich** [comment]
Using `metadataInput` is the currently expected way of getting at explicitly specified values; however, accessing the name as a Terraform reference should be possible. It isn't currently because of https://github.com/hashicorp/terraform-cdk/blob/fe4a10f1f48bb6dfd2d5e1f34f4d4fe61e0c037b/packages/cdktf-cli/lib/get/generator/emitter/attributes-emitter.ts#L225.
This should get fixed with #525.

**Comment by DanielMSchmidt** [comment]
This should be fixed in 0.7, metadata is a single item block
